### ISSUE TITLE
Cannot find libjasper, not listed by ldd

### ISSUE THREAD

**Issue by gusmd** [probonopd/linuxdeployqt#109]
When running linuxdeployqt on my project, I get the following:

```
Not using FHS-like mode
app-binary: "/build/linux/ui/ui"
appDirPath: "/build/linux/ui"
relativeBinPath: "ui"
ERROR: ldd outputLine: "libjasper.so.1 => not found"
ERROR: Please ensure that all libraries can be found by ldd. Aborting.

```

I'm using Qt 5.9 beta3, in a Ubuntu 17.04 fresh installation. `ldd` does not list libjasper as a dependency.

Anything I could be doing wrong?

**Comment by probonopd** [comment]
Hi @gusmd welcome to AppImage!

First of all, I would not recommend to use Ubuntu 17.04 at this point since it is very new and the resulting binaries will not run on pre-2017 systems. Much better to use Ubuntu 14.04 Trusty at this point.

Next, where does `libjasper.so.1` live on your build system? `linuxdeployqt`is using `ldd`to find libraries.

Assuming it lives in `/usr/lib/x86_64/jasper/`, try

```
export LD_LIBRARY_PATH=/usr/lib/x86_64/jasper/
./linuxdeployqt*.AppImage ...
```

**Comment by probonopd** [comment]
Please retry with the latest continuous build, is this still an issue for you?

**Comment by Zaraka** [comment]
I just get this error on latest Fedora 26
```
ERROR: ldd outputLine: "libjasper.so.1 => not found"                                                                                                                                                                                         
ERROR: Please ensure that all libraries can be found by ldd. Aborting
```
*dnf list jasper jasper-devel*
```
Installed Packages
jasper.x86_64                                                                                                         2.0.12-1.fc26                                                                                                   @fedora
jasper-devel.x86_64                                                                                                   2.0.12-1.fc26                                                                                                   @fedora
```

so i ran *locate jasper* and here's the output

```
/usr/lib/libjasper.so
/usr/lib/libjasper.so.4
/usr/lib/libjasper.so.4.0.0
/usr/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-jasper.so
/usr/lib64/libjasper.so
/usr/lib64/libjasper.so.4
/usr/lib64/libjasper.so.4.0.0
/usr/lib64/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-jasper.so
```

And now I'm scratching head, it looks like .so.1 is unavailable in Fedora 26.
The application itself is Qt 5.9.1 compiling and running works ok on this system

**Comment by probonopd** [comment]
@Zaraka which software are you trying to turn into an AppImage?

**Comment by Zaraka** [comment]
@probonopd Qt app with a single dependency OpenCV. But jasper is used in Qt itself isn't it?

**Comment by probonopd** [comment]
`cd` to your AppDir, then run `grep -r libjasper .`. If your app is publicly available, I can have a look at it.

**Comment by randinterval** [comment]
Hi, I'm having same issue - I'm on Ubuntu 17.04

saad@Saad-Home:~/Desktop/app$ ./app.load App
Not using FHS-like mode
app-binary: "/home/saad/Desktop/app/App"
appDirPath: "/home/saad/Desktop/app"
relativeBinPath: "App"
ERROR: ldd outputLine: "libjasper.so.1 => not found"
ERROR: Please ensure that all libraries can be found by ldd. Aborting.

**Comment by randinterval** [comment]
I figured out why this is happening. Apparently,  jasper was removed in Debain (Ubuntu is debian based), hence the error. I was able to find a standalone package for jasper, but decided to just download previous version of Ubuntu trusty and build there as I'm going to deploy on 14.04, so makes no sense to package on 17.04

**Comment by probonopd** [comment]
If you create the AppImage properly on 14.04, then it should run on subsequent versions like 17.04 as well. If it doesn't, I'd consider it as a bug.

**Comment by randinterval** [comment]
@probonopd yes, I understand that - I was trying to build it on 17.04, and my target machines which will actually run the application are on 14.04, so figured it would be better to build on 14.04.
PS: Thanks for the great tool! Very useful.
